src/pymor/models/iosys.py

```diff
@@ -750,70 +855,48 @@ def impulse_resp(self, mu=None, return_solution=False):
+        n = (self.num_values if self.num_values else self.time_stepper.estimate_time_step_count(0, self.T)) + 1
+        output = np.empty((n, self.dim_output, self.dim_input))
+        if return_solution:
+            solution = []
 
-        # solution computation
-        common_solver_opts = dict(
-            operator=-self.A,
-            mass=None if isinstance(self.E, IdentityOperator) else self.E,
-            initial_time=0,
-            end_time=self.T,
-            num_values=self.num_values,
-        )
         if self.sampling_time == 0:
-            Xs0 = self.E.apply_inverse(self.B.as_range_array(mu=mu), mu=mu)
-            Xs = tuple(
-                self.time_stepper.solve(
-                    rhs=None,
-                    initial_data=X0,
-                    mu=mu,
-                    **common_solver_opts,
-                )
-                for X0 in Xs0
-            )
-        else:
-            rhs = LinearInputOperator(self.B)
-            Xs = []
-            for i in range(self.dim_input):
-                def input_i(t):
+            initial_data = self.E.apply_inverse(self.B.as_range_array(mu=mu), mu=mu)
+
+        for i in range(self.dim_input):
+            if self.sampling_time == 0:
+                data = self.with_(initial_data=initial_data[i]).compute(input=np.zeros(self.dim_input),
+                                                                        output=True,
+                                                                        solution=return_solution,
+                                                                        mu=mu)
+            else:
+                def input(t):
+                    e = np.zeros(self.dim_input)
                     if t == 0:
-                        e_i = np.zeros(self.dim_input)
-                        e_i[i] = 1/self.sampling_time
-                        return e_i
-                    return np.zeros(self.dim_input)
-                Xs.append(
-                    self.time_stepper.solve(
-                        rhs=rhs,
-                        initial_data=self.solution_space.zeros(1),
-                        mu=mu.with_(input=GenericFunction(input_i, shape_range=(self.dim_input,))),
-                        **common_solver_opts,
-                    )
-                )
-            Xs = tuple(Xs)
-
-        # output computation
-        y = np.empty((len(Xs[0]), self.dim_output, self.dim_input))
-        for i, X in enumerate(Xs):
-            y[:, :, i] = self.C.apply(X, mu=mu).to_numpy()
-        if self.sampling_time > 0 and not isinstance(self.D, ZeroOperator):
-            y[0] += to_matrix(self.D, mu=mu, format='dense') / self.sampling_time
+                        e[i] =  1 * self.sampling_time if self.sampling_time > 0 else 1
+                    return e
+                input = GenericFunction(mapping=input, shape_range=(self.dim_input,))
+                data = self.compute(input=input, output=True, solution=return_solution, mu=mu)
```

Comment:
The initial condition could be set to something nonzero.

```suggestion
                data = self.with_(initial_data=self.solution_space.zeros(1)).compute(input=input, output=True, solution=return_solution, mu=mu)
```